lib/extensions/virtual_total.rb

```diff
@@ -93,17 +93,33 @@ def define_virtual_aggregate_method(name, relation, method_name, column)
     def virtual_aggregate_arel(reflection, method_name, column)
       return unless reflection && reflection.macro == :has_many && !reflection.options[:through]
       lambda do |t|
+        query = if reflection.scope
+                  reflection.klass.instance_exec(nil, &reflection.scope)
+                else
+                  reflection.klass.all
+                end
+
+        # ordering will probably screw up aggregations, so clear this out from
+        # any calls
+        #
+        # only clear this out if this isn't a `:size` call as well, since doing
+        # a COUNT(*) will allow any ORDER BY to still work properly.  This is
+        # to avoid any possible edge cases by clearing out the order clause.
+        query.order_values = [] if method_name != :size
```

Comment:
interesting, I would have expected the opposite: For `size` to be the case where the `order_values` were not needed.

But this seems to make sense

Reply:
Well, with `:size`, you are doing an `Arel.star.count` (see below), meaning all of columns will be in the query, and the SQL statement with the `ORDER BY` will work with whatever column(s) are being ordered on, even though you don't need it for an aggregate.

Hopefully we don't start doing a `JOIN` in these `virtual_attributes` (though, you and me both have talked about why that would be a good thing over nested selects), since that is where this would just need to be removed, because then we would be probably getting order by columns from other tables.